DatabaseOperations/Services/ConnectionStringService.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace DatabaseOperations.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text.RegularExpressions;
+    using ConnectionRules;
+    using DataTransferObjects;
+    using Interfaces;
+
+    internal static class ConnectionStringService
+    {
+        private static readonly char[] SplitArray = { ';' };
+
+        private static readonly IList<IConnectionRule> ConnectionRules = new List<IConnectionRule>
+        {
+            new ApplicationConnectionRule(),
+            new ConnectTimeoutConnectionRule(),
+            new ConnectionTimeoutConnectionRule(),
+            new DataSourceConnectionRule(),
+            new ServerConnectionRule(),
+            new AddressConnectionRule(),
+            new AbbreviatedAddressConnectionRule(),
+            new NetworkAddressConnectionRule(),
+            new CatalogConnectionRule(),
+            new DatabaseConnectionRule(),
+            new SecurityConnectionRule(),
+            new TrustedConnectionRule(),
+            new PasswordConnectionRule(),
+            new AbbreviatedPasswordConnectionRule(),
+            new UserConnectionRule()
+        };
+
+        internal static ConnectionProperties ExtractConnectionParameters(string connectionString)
+        {
+            if (string.IsNullOrWhiteSpace(connectionString)) return new ConnectionProperties();
+
+            string[] itemArray = connectionString.Split(SplitArray, StringSplitOptions.RemoveEmptyEntries);
+
+            ConnectionProperties connectionProperties = ProcessItemArray(itemArray);
+            connectionProperties.ConnectionString = connectionString;
```

Comment:
This needs changing.

The connection string can be set using the next method:
`UpdateConnectionString()`